src/components/NewCommentForm.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useCallback, useState } from 'react';
+import cn from 'classnames';
+import { client } from '../utils/fetchClient';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+
+interface Props {
+  onCommentAdd: (comms: Comment[] | ((prev: Comment[]) => Comment[])) => void
+  currentPost: Post
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  onCommentAdd,
+  currentPost,
+}) => {
+  const [name, setName] = useState('');
+  const [hasNameError, setHasNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [hasEmailError, setHasEmailError] = useState(false);
+  const [text, setText] = useState('');
+  const [hasTextError, setHasTextError] = useState(false);
+  const [isFormLoading, setIsFormLoading] = useState(false);
+
+  const commentAddHandler = useCallback((
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    e.preventDefault();
+
+    if (!name || !email || !text) {
+      setHasNameError(!name);
+      setHasEmailError(!email);
+      setHasTextError(!text);
+
+      return;
+    }
+
+    setIsFormLoading(true);
+
+    client.post<Comment>('/comments', {
+      postId: currentPost.id,
+      name,
+      email,
+      body: text,
+    }).then(post => onCommentAdd(posts => [...posts, post]))
+      .finally(() => {
+        setIsFormLoading(false);
+        setText('');
+      });
+  }, [name, email, text]);
+
+  const formClearHandler = useCallback(() => {
+    setEmail('');
+    setName('');
+    setText('');
+    setHasNameError(false);
+    setHasEmailError(false);
+    setHasTextError(false);
+  }, []);
+
+  return (
+    <form data-cy="NewCommentForm">
+      <div className="field" data-cy="NameField">
+        <label className="label" htmlFor="comment-author-name">
+          Author Name
+        </label>
+
+        <div className="control has-icons-left has-icons-right">
+          <input
+            type="text"
+            name="name"
+            id="comment-author-name"
+            placeholder="Name Surname"
+            className={cn(
+              'input',
+              { 'is-danger': hasNameError },
+            )}
+            value={name}
+            onChange={(e) => {
+              setName(e.target.value);
+              setHasNameError(false);
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function). Fix all cases